async_fuse/src/memfs/node.rs

```diff
@@ -1050,12 +1094,12 @@ impl Node {
     // File only methods
 
     /// Get file data
-    pub fn get_file_data(&self) -> &Vec<u8> {
+    pub fn get_file_data(&self, offset: usize, len: usize) -> Vec<IoMemBlock> {
```

Comment:
A little confused here, some element in the `get_file_cache` return value could be None, right? Means there's no buffer of this range in the Cache. If so, this `get_file_data` function may only return part of the file data, is it expected?

Reply:
Just after we call the `load_data`, then the data in Cache is deleted because of Cache capability, then we call this `get_file_data`, shouldn't we read the data from file, update Cache and return?

Reply:
Any data loaded in the `load_data` should be in the front end of the priority queue, which should not be deleted. 

We may load partial data from the file instead of the whole requests for the user, in case that the cache has holded part of the requests range. So we can't return the data from file directly. In short the operation is separated into two steps:

1. fill the cache.
2. read data in the cache.

Reply:
I see, I mean what if we first call `load_data`, the data is loaded in Cache, but before we call `get_file_data`, the data in Cache is evicted (Maybe because of other write operations), then right now we call `get_file_data`, wouldn't it be a problem? (some of the data is lost)